### ISSUE TITLE
False negative: `no-unused-vars` doesn't check variables introduced in `for-of`, `for-in`

### ISSUE THREAD

**Issue by magurotuna** [denoland/deno_lint#697]
### Lint Name

no-unused-vars

### Code Snippet

```ts
for (const x of xs) {
  // noop
}

for await (const x of xs) {
  // noop
}

for (const x in xs) {
  // noop
}
```

### Expected Result

The linter spots all `x` as errors

### Actual Result

No error spotted

### Additional Info

None

### Version

deno 1.10.1 (release, x86_64-unknown-linux-gnu)
v8 9.1.269.27
typescript 4.2.2
